packages/blade/src/components/Drawer/_decisions/decisions.md

```diff
@@ -0,0 +1,147 @@
+# Drawer
+
+A drawer is a panel that slides in mostly from right side of the screen over the existing content in the viewport. It helps in providing additional details or context and can also be used to promote product features or new products.
+
+<img width="500px" src="./2024-02-07-20-04-07.png" alt="Drawer Figma Skeleton">
+
+## Design
+
+- [Drawer - Figma Design](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?node-id=78667%3A66663&mode=dev)
+
+## Proposed API
+
+```jsx
+<Drawer
+  isOpen={}
+  onDismiss={() => {}}
+  showOverlay
+>
+  <DrawerHeader
+    title=""
+    subtitle=""
+    leading={<DrawerHeaderIcon />}
+    titleSuffix={<DrawerHeaderBadge />}
+    trailing={
+      <>
+        <DrawerHeaderLink />
+        <DrawerHeaderButton />
+      </>
+    }
+  />
+  <DrawerBody>
+    <Slot />
+  </DrawerBody>
+<Drawer>
+```
+
+<details>
+<summary><b>Full Usage Example</b></summary>
+
+```jsx
+const MyCuteDrawer = () => {
+  const [showDrawer, setShowDrawer] = React.useState(false);
+  return (
+    <Box>
+      <Button onClick={() => setShowDrawer(true)}>Open Drawer</Button>
+      <Drawer
+        isOpen={showDrawer}
+        onDismiss={() => {
+          setShowDrawer(false);
+        }}
+      >
+        <DrawerHeader
+          title="Announcements"
+        />
+        <DrawerBody>
+          <FTXAnnouncement />
+          <RazorpayOnePromotions />
+          <CatPictures />
+        </DrawerBody>
+      <Drawer>
+    </Box>
+  )
+}
+
+```
+
+</details>
+
+### Props
+
+```ts
+type DrawerProps = {
+  /**
+   * Controlled state of drawer open or not
+   */
+  isOpen: boolean;
+
+  /**
+   * Dismiss handler
+   */
+  onDismiss: () => void;
+
+  /**
+   * Show or hide overlay.
+   *
+   * Also decides if clicking outside on overlay closes the drawer or not
+   */
+  showOverlay?: boolean;
+};
+
+type DrawerHeaderProps = {
+  /**
+   * Title of the Drawer
+   */
+  title: string;
+
+  /**
+   * Subtitle of the Drawer
+   */
+  subtitle?: string;
+
+  /**
+   * Leading element
+   *
+   * DrawerHeaderIcon or DrawerHeaderAsset
+   */
+  leading?: ReactNode;
+
+  /**
+   * Title suffix element
+   *
+   * DrawerHeaderBadge
+   */
+  titleSuffix?: ReactNode;
+
+  /**
+   * Title trailing element
+   *
+   * DrawerHeaderLink, DrawerHeaderButton[]
+   */
+  trailing?: ReactNode;
+};
+```
+
+Other supporting wrapper components for trailing and leading space-
+
+- DrawerHeaderBadge
+- DrawerHeaderIcon
+- DrawerHeaderAsset
+- DrawerHeaderLink
+- DrawerHeaderButton
+
+_No alternate APIs were considered because Drawer is closer to Modal on overall meaning and API perspective so made sense to go with API that is closer to Modal, also all DS Drawer components I referenced have similar API_
+
+## Accessibility
+
+- **Aria Attributes:** Drawer will have `aria-modal="true"` and `role="dialog"` and will be treated as modal for voiceover users.
+- **Focus Handling:** Ensure Close Icon is focussable
```

Comment:
Like Modal & Bottomsheet, should we also expose i`nitialFocusRef`?

Reply:
Yes I think we should. Drawer will also have similar usecases like Modal and BottomSheet so requirement will anyway come

Reply:
Add keyboard handling details, e.g, `Esc` should close the drawer.

Reply:
done